### DISCUSSION TITLE
Why does TAB key show 8 spaces length?

### DISCUSSION THREAD

**Discussion by sgon00** [neovide/neovide#2911] [Q&A] [answered]
Hi, I am new to neovide. For neovim in terminal, TAB key just shows 2 spaces. But for neovide, TAB keys shows 8 spaces long. Why? I am fine with 2 spaces or 4 spaces long. But 8 spaces long is too much, how to configure neovide for TAB length? Thanks a lot.

**Comment by fredizzimo** [comment, marked as answer]
The default tab size of Neovim is 8 https://neovim.io/doc/user/options.html#'tabstop', and so is the shiftwidth https://neovim.io/doc/user/options.html#'shiftwidth'

Most filetype plugins redefine it common settings for that language. So that's probably why you see 2 in some places. But the terminal Neovim and Neovide should be the same for the same file.

You can also configure the settings manually, or use an [EditorConfig](https://editorconfig.org/) file.

**Reply by sgon00** [reply]
Thanks a lot for your quick reply.  You said that `tabstop` value should be the same in terminal neovim and neovide. But that is not the case. I am running neovim + nvchad + neovide in Xubuntu 24.04 (xfce x11). Kinda clean installation so far. I didn't install any other plugins and configure too much. I tried `vim`, `neovim` and `neovide`. Three of them are using all different `tabstop` values. 

For `toml` file, 
* `vim` is using 4 tabstop. 
* `terminal neovim` is using 2 tabstop.
* `neovide` is using 8 tabstop.

I found the value by executing `:set tabstop`. I have explicitly set tabstop=4 in `.vimrc`. I didn't set anything explicitly in `neovim` and `neovide`. They are just using different values. So why and what parameters should I set and where should I set those values? I think tabstop value should be different per filetype. And as what you said, `neovide` should be using the same value as `terminal neovim`.

**Reply by sgon00** [reply]
I found the cause and I think it's a bug in NvChad. I don't know how to delete a question here so I marked this as the answer. Basically, the problem happens when I open multiple files in the command line. The behavior is the same in both terminal and neovide. Sorry about it.
